src/dom-wrapper.ts

```diff
@@ -10,19 +10,23 @@ export class DOMWrapper<ElementType extends Element> {
     this.element = element
   }
 
-  classes(className?: string) {
+  classes(): string[]
+  classes(className: string): boolean
+  classes(className?: string): string[] | boolean {
     const classes = this.element.classList
 
     if (className) return classes.contains(className)
 
     return Array.from(classes)
   }
 
-  attributes(key?: string) {
+  attributes(): { [key: string]: string }
+  attributes(key: string): string
+  attributes(key?: string): { [key: string]: string } | string {
     const attributes = this.element.attributes
-    const attributeMap = {}
+    const attributeMap: { [key: string]: string } = {}
     for (let i = 0; i < attributes.length; i++) {
-      const att = attributes.item(i)
+      const att = attributes.item(i)!
```

Comment:
If the attribute is missing, won't `att.localName` give "could not find `localName` of undefined?

Reply:
we are iterating the attribute list, so I don't think is possible this is `undefined` at runtime

Reply:
Right 🤦

Reply:
I had to double check when I was reviewing, probably a forof would be a bit more explicit, but that's a bit nit picking :D

Reply:
Yeah I agree. I had to double check too when I added this `!`.

Reply:
It looks like we all agree, so I pushed an extra commit with a refactoring to use forof instead.